### PR TITLE
feat: Add Privacy, Cookie Policy and Terms Preference

### LINKED ISSUE #1218: Add Privacy Policy
### Feature Request

**Feature Description**

User should be able to view Privacy Policy in Device Settings-> Legal -> Privacy -> Open Eventyay privacy policy in browser.
   
**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by Masquerade0097**
Fixes #1218 

Checklist:

- [x] I have checked for PMD and check-style issues 
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream development branch.

Changes: 

Screenshots for the change:
kill

iii

### CODE REVIEW

**Comment by Masquerade0097** [comment]
@iamareebjamal @sridharjajoo @srv-twry Please review.

**Comment by sridharjajoo** [comment]
Commit message can be changed to "Add Privacy Policy Preference"

File: app/src/main/java/org/fossasia/openevent/app/core/settings/SettingsFragment.java [hunk 1]
```diff
@@ -52,6 +52,14 @@ public void onCreatePreferencesFix(@Nullable Bundle bundle, String rootKey) {
             }
             return true;
         });
+
+        findPreference(getString(R.string.privacy_policy_key)).setOnPreferenceClickListener(preference -> {
+            String url = "https://eventyay.com/privacy-policy/";
```
[the remark this answers is not in the document]
Comment (Masquerade0097) [reply app/src/main/java/org/fossasia/openevent/app/core/settings/SettingsFragment.java]: Why?

[the remark this answers is not in the document]
Comment (Masquerade0097) [reply app/src/main/java/org/fossasia/openevent/app/core/settings/SettingsFragment.java]: Okay, I didn't notice it was there in Login screen as well.

**Comment by Masquerade0097** [comment]
@mishuvs I have done the changes. Please review.

**Comment by Masquerade0097** [comment]
@mishuvs EventBrite's signup page contains an option to access the entire Device Settings not just these 3 options. This should be handled in a separate PR.

**Comment by mishuvs** [comment]
I am referring to the three small links above the Sign Up button, not Device Settings. IMO Device Settings is not needed in Auth Fragments. Most apps show settings options only after the user logs in. Otherwise LGTM.

**Comment by Masquerade0097** [comment]
@mishuvs Please review the updated PR.

File: app/src/main/java/org/fossasia/openevent/app/common/Constants.java [hunk 2]
```diff
@@ -21,6 +21,9 @@
     public static final String PREF_PAYMENT_ACCEPT_ONSITE = "accept_onsite";
     public static final String PREF_PAYMENT_ONSITE_DETAILS = "onsite_details";
     public static final String PREF_PAYMENT_COUNTRY = "key";
+    public static final String PRIVACY_POLICY_URL = "https://eventyay.com/privacy-policy/";
+    public static final String COOKIE_POLICY_URL = "https://eventyay.com/cookie-policy/";
+    public static final String TERMS_OF_USE_URL = "https://eventyay.com/terms/";
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/common/Constants.java]: Add these locally

File: app/src/main/java/org/fossasia/openevent/app/core/auth/signup/SignUpFragment.java [hunk 3]
```diff
@@ -42,6 +44,9 @@
     private Validator validator;
     private SharedViewModel sharedViewModel;
 
+    public final String PRIVACY_POLICY_URL = "https://eventyay.com/privacy-policy/";
+    public final String TERMS_OF_USE_URL = "https://eventyay.com/terms/";
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/auth/signup/SignUpFragment.java]: Reuse

**Comment by Masquerade0097** [comment]
@iamareebjamal Please review.

**Comment by Masquerade0097** [comment]
@iamareebjamal Please review.

**Comment by Masquerade0097** [comment]
@mishuvs I might have checked the box by mistake. Anyways, violations are resolved now.

File: app/src/main/java/org/fossasia/openevent/app/core/auth/signup/SignUpFragment.java [hunk 4]
```diff
@@ -42,6 +44,10 @@
     private Validator validator;
     private SharedViewModel sharedViewModel;
 
+    public final String EVENTYAY_BASE_URL = "https://eventyay.com";
+    public final String PRIVACY_POLICY_URL = EVENTYAY_BASE_URL + "/privacy-policy/";
+    public final String TERMS_OF_USE_URL = EVENTYAY_BASE_URL + "/terms/";
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/auth/signup/SignUpFragment.java]: Still duplication

File: app/src/main/java/org/fossasia/openevent/app/core/settings/LegalPreferenceFragment.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+import org.fossasia.openevent.app.R;
+import org.fossasia.openevent.app.common.Constants;
+import org.fossasia.openevent.app.ui.ViewUtils;
+
+public class LegalPreferenceFragment extends PreferenceFragmentCompat {
+
+    public static final String EVENTYAY_BASE_URL = "https://eventyay.com/";
+    public final String privacyPolicyUrl = "privacy-policy/";
+    public final String cookiePolicyUrl = "cookie-policy/";
+    public final String termsOfUseUrl = "terms/";
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/settings/LegalPreferenceFragment.java]: All three of these strings why you just can't use a single file to declare and reuse in other file. Duplication is not good

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/app/core/settings/LegalPreferenceFragment.java]: Yes, but in one file, not in both

File: app/src/main/java/org/fossasia/openevent/app/core/settings/LegalPreferenceFragment.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+import org.fossasia.openevent.app.R;
+import org.fossasia.openevent.app.common.Constants;
+import org.fossasia.openevent.app.ui.ViewUtils;
+
+public class LegalPreferenceFragment extends PreferenceFragmentCompat {
+
+    public static final String EVENTYAY_BASE_URL = "https://eventyay.com/";
+    public static final String PRIVACY_POLICY_URL = "privacy-policy/";
+    public final String cookiePolicyUrl = "cookie-policy/";
+    public static final String TERMS_OF_USE_URL = "terms/";
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/settings/LegalPreferenceFragment.java]: Write full URL in each field, the duplication I was talking about was of the fields, remove base url and make all URLs fully qualified. Make cookiePolicyUrl static as well

**Pull request merged**